File: scripts/capture_screens.py
from playwright.sync_api import sync_playwright

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.set_viewport_size({"width": 1280, "height": 800})
        
        # Capture Login
        page.goto('http://localhost:3002/auth/login.html')
        page.wait_for_load_state('networkidle')
        page.screenshot(path='demo-video/public/captures/scene4-login.png')

        # Capture Client Portal (from showcase)
        page.goto('http://localhost:3002/showcase.html')
        page.wait_for_load_state('networkidle')
        page.evaluate("document.querySelector('.tab-btn:nth-child(3)').click()")
        page.wait_for_timeout(1000)
        # Capture just the preview container
        clip = page.locator('.preview-container').bounding_box()
        if clip:
            page.screenshot(path='demo-video/public/captures/scene2-calc.png', clip=clip)
        else:
            page.screenshot(path='demo-video/public/captures/scene2-calc.png')

        # Capture Admin Dashboard (from showcase)
        page.evaluate("document.querySelector('.tab-btn:nth-child(1)').click()")
        page.wait_for_timeout(1000)
        clip = page.locator('.preview-container').bounding_box()
        if clip:
            page.screenshot(path='demo-video/public/captures/scene3-dashboard.png', clip=clip)
        else:
            page.screenshot(path='demo-video/public/captures/scene3-dashboard.png')

        browser.close()

if __name__ == '__main__':
    run()

File: scripts/capture_screens.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  await page.setViewportSize({ width: 1280, height: 800 });

  // Capture Login
  await page.goto('http://localhost:3002/auth/login.html', { waitUntil: 'networkidle' });
  await page.screenshot({ path: 'demo-video/public/captures/scene4-login.png' });

  // Capture Client Portal
  await page.goto('http://localhost:3002/showcase.html', { waitUntil: 'networkidle' });
  await page.click('.tab-btn:nth-child(3)');
  await page.waitForTimeout(1000);
  
  let clip = await page.evaluate(() => {
    const el = document.querySelector('.preview-container');
    if (!el) return null;
    const rect = el.getBoundingClientRect();
    return { x: rect.x, y: rect.y, width: rect.width, height: rect.height };
  });
  
  if (clip) {
    await page.screenshot({ path: 'demo-video/public/captures/scene2-calc.png', clip });
  } else {
    await page.screenshot({ path: 'demo-video/public/captures/scene2-calc.png' });
  }

  // Capture Admin Dashboard
  await page.click('.tab-btn:nth-child(1)');
  await page.waitForTimeout(1000);
  clip = await page.evaluate(() => {
    const el = document.querySelector('.preview-container');
    if (!el) return null;
    const rect = el.getBoundingClientRect();
    return { x: rect.x, y: rect.y, width: rect.width, height: rect.height };
  });

  if (clip) {
    await page.screenshot({ path: 'demo-video/public/captures/scene3-dashboard.png', clip });
  } else {
    await page.screenshot({ path: 'demo-video/public/captures/scene3-dashboard.png' });
  }

  await browser.close();
})();
